garrysmod/gamemodes/base/entities/entities/base_trigger.lua

```diff
@@ -0,0 +1,95 @@
+
+ENT.Base = "base_entity"
+ENT.Type = "trigger"
+
+--[[---------------------------------------------------------
+	Name: Initialize
+-----------------------------------------------------------]]
+function ENT:Initialize()
+	self:SetTrigger(true)
+	self.Entities = {}
+end
+
+--[[---------------------------------------------------------
+	Name: IsTouchedBy
+-----------------------------------------------------------]]
+function ENT:IsTouchedBy(ent)
+	return self.Entities[ent]
+end
+
+--[[---------------------------------------------------------
+	Name: StartTouch
+	Desc: Called when a entity starts touching us
+-----------------------------------------------------------]]
+function ENT:StartTouch(ent)
+	if ( !self:PassesTriggerFilters(ent) ) then return end
+	self.Entities[ent] = ent
+	
+	self:Input("OnStartTouch", self, ent)
+end
+
+--[[---------------------------------------------------------
+	Name: EndTouch
+	Desc: Called when a entity stops touching us
+-----------------------------------------------------------]]
+function ENT:EndTouch(ent)
+	if ( !self:IsTouchedBy(ent) ) then return end
+	self.Entities[ent] = nil
+	
+	self:Input("OnEndTouch", self, ent)
+
+	local bFoundOtherTouchee = false
+	local iSize = #self.Entities
+	for i=iSize, 0, -1 do 
```

Comment:
Additionally, you cannot use the length operator to accurately find the size of a table that is not both numeric and sequential.